test_selenium/profile/admin/test_profile_indicator_admin.py

```diff
@@ -104,9 +104,34 @@ def test_enable_linear_scrubber_option(self):
         assert linear_scrubber_checkbox.is_selected() == False
         assert linear_scrubber_checkbox.is_enabled() == False
 
-        variable_field.select_by_value(str(self.indicator2.id))
+        variable_field.select_by_value(str(self.indicator4.id))
         assert linear_scrubber_checkbox.is_enabled() == True
         linear_scrubber_checkbox.click()
         href = "/".join(link_to_group.get_attribute("href").split("/")[3:])
         # Check if link in warning alert also works for private variables
-        assert href == f"admin/datasets/group/{self.group2.id}/change/"
+        assert href == f"admin/datasets/group/{self.group4.id}/change/"
+
+        # Save and continue and check if setting is saved
+        subcategory_element = main_div.find_element(by=By.CSS_SELECTOR, value="select#id_subcategory")
+        subcategory_field = Select(subcategory_element)
+        subcategory_field.select_by_value(str(self.indicator_subcategory.id))
+
+        choropleth_method_element = main_div.find_element(by=By.CSS_SELECTOR, value="select#id_choropleth_method")
+        choropleth_method_field = Select(choropleth_method_element)
+        choropleth_method_field.select_by_value(str(self.choropleth_method.id))
+
+        main_div.find_element_by_css_selector(
+            ".submit-row input[value='Save and continue editing']"
+        ).click()
+
+        WebDriverWait(self.selenium, 10).until(
+            EC.visibility_of_element_located((By.TAG_NAME, "h1"))
+        )
+
+        main_div = self.get_element("content")
+        header_text = main_div.find_element(by=By.TAG_NAME, value="h1").text
+        assert header_text == "Change profile indicator"
```

Comment:
This is a standard check to ensure the correct page is loaded and ready to work. I wondered if it would be useful to move this code block to a separate function so that in the test all we need is something similar to

`self.wait_until("h1", "Change profile indicator")`

What do you think?

Reply:
agreed 
Updated 👍